src/cider/nrepl/middleware/debug.clj

```diff
@@ -438,12 +438,28 @@
             (eval form)
             (alter-meta! v assoc ::instrumented instrumented)))))))
 
+(def step-in-blacklist
+  "A set of names or regexes specifying namespaces that should not be stepped in
+  to. By default, this contains only \"clojure.core\"."
+  (atom #{"clojure.core"}))
+
+(defn blacklisted-for-step-in?
```

Comment:
As implemented right now, this only applies to step-in. If you want to manually navigate into "clojure/core.clj" and instrument a function, we don't prevent that. Do you think we should?